misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -311,8 +312,25 @@ public FunctionDefinitionNode transform(FunctionDefinitionNode functionDefinitio
         }
         NodeList<Node> relativeResourcePath = formatNodeList(functionDefinitionNode.relativeResourcePath(), 0, 0, 0, 0);
         FunctionSignatureNode functionSignatureNode = formatNode(functionDefinitionNode.functionSignature(), 1, 0);
+        int trailingNL = env.trailingNL;
+        if (isMemberOfScope(functionDefinitionNode)) {
+            ChildNodeList parentChildren = functionDefinitionNode.parent().children();
```

Comment:
This is not the correct place to handle `itemTrailingNL` for service and class members. I think we need to handle them at the parent level itself in [1] and [2]. I expect the implementation to be similar to `formatModuleMembers()` we have implemented in [3] for formatting the module members.

[1]. https://github.com/ballerina-platform/ballerina-lang/blob/2752ae1d5f93a9f99684812125f0530dd614ebe7/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java#L3465
[2]. https://github.com/ballerina-platform/ballerina-lang/blob/2752ae1d5f93a9f99684812125f0530dd614ebe7/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java#L703
[3]. https://github.com/ballerina-platform/ballerina-lang/blob/2752ae1d5f93a9f99684812125f0530dd614ebe7/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java#L291

Reply:
Fixed